core/src/main/java/hudson/security/SecurityRealm.java

```diff
@@ -484,6 +488,56 @@ public Filter createFilter(FilterConfig filterConfig) {
      */
     public static final SecurityRealm NO_AUTHENTICATION = new None();
 
+    /**
+     * Perform a calculation where we should go back after sucessfull login
+     *
+     * @return Encoded URI where we should go back after sucessfull login
+     *         or "/" if no way back or an issue occurred
+     *
+     * @since TODO
+     */
+    @Restricted(DoNotUse.class)
+    public static String getFrom() {
+        String from = null, returnValue = null;
+        final StaplerRequest request = Stapler.getCurrentRequest();
+
+        // Try to obtain a return point either from the Session
+        // or from the QueryParameter in this order
+        if (request != null
+                && request.getSession(false) != null) {
+            from = (String) request.getSession().getAttribute("from");
+        } else if (request != null) {
+            from = request.getParameter("from");
+        }
+
+        // If entry point was not found, try to deduce it from the request URI
+        // except pages related to login process
+        if (from == null
+                && request != null
+                && request.getRequestURI() != null
+                && request.getRequestURI().equals("/loginError")
+                && request.getRequestURI().equals("/login")) {
```

Comment:
@oleg-nenashev Ough, you are right my bad, I've modified this part of conditions from compareTo() to equals(), but reverse the condition, fixing...